executor/src/witgen/machines/profiling.rs

```diff
@@ -73,10 +73,14 @@ pub fn reset_and_print_profile_summary() {
             });
 
             // Finish the execution of the currently running machine.
-            let duration = time.duration_since(current_time).unwrap();
-            *time_by_machine
-                .entry(current_machine_id)
-                .or_insert(Duration::default()) += duration;
+            // This should almost always return Ok(), but it could be that the system
```

Comment:
Why don't we use a proper non-decreasing clock instead? We don't care about real time here, only durations.